python/private/pypi/pkg_aliases.bzl

```diff
@@ -132,3 +145,253 @@ def pkg_aliases(
+    """Get the filename config settings.
+
+    Args:
+        filename: the distribution filename (can be a whl or an sdist).
+        target_platforms: list[str], target platforms in "{abi}_{os}_{cpu}" format.
+        glibc_versions: list[tuple[int, int]], list of versions.
+        muslc_versions: list[tuple[int, int]], list of versions.
+        osx_versions: list[tuple[int, int]], list of versions.
+        python_version: the python version to generate the config_settings for.
+        non_whl_prefix: the prefix of the config setting when the whl we don't have
+            a filename ending with ".whl".
+
+    Returns:
+        A tuple:
+         * A list of config settings that are generated by ./pip_config_settings.bzl
+         * The list of default version settings.
+    """
+    prefixes = []
+    suffixes = []
+    setting_supported_versions = {}
+
+    if filename.endswith(".whl"):
+        if (0, 0) in glibc_versions:
+            fail("Invalid version in 'glibc_versions': cannot specify (0, 0) as a value")
+        if (0, 0) in muslc_versions:
+            fail("Invalid version in 'muslc_versions': cannot specify (0, 0) as a value")
+        if (0, 0) in osx_versions:
+            fail("Invalid version in 'osx_versions': cannot specify (0, 0) as a value")
+
+        glibc_versions = sorted(glibc_versions)
+        muslc_versions = sorted(muslc_versions)
+        osx_versions = sorted(osx_versions)
+
+        parsed = parse_whl_name(filename)
+        if parsed.python_tag == "py2.py3":
+            py = "py"
+        elif parsed.python_tag.startswith("cp"):
+            py = "cp3x"
+        else:
+            py = "py3"
+
+        if parsed.abi_tag.startswith("cp"):
+            abi = "cp"
+        else:
+            abi = parsed.abi_tag
+
+        if parsed.platform_tag == "any":
+            prefixes = ["_{}_{}_any".format(py, abi)]
+        else:
+            prefixes = ["_{}_{}".format(py, abi)]
+            suffixes = _whl_config_setting_suffixes(
+                platform_tag = parsed.platform_tag,
+                glibc_versions = glibc_versions,
+                muslc_versions = muslc_versions,
+                osx_versions = osx_versions,
+                setting_supported_versions = setting_supported_versions,
+            )
+    else:
+        prefixes = [""] if not non_whl_prefix else ["_" + non_whl_prefix]
+
+    versioned = {
+        ":is_cp{}{}_{}".format(python_version, p, suffix): {
+            version: ":is_cp{}{}_{}".format(python_version, p, setting)
+            for version, setting in versions.items()
+        }
+        for p in prefixes
+        for suffix, versions in setting_supported_versions.items()
+    }
+
+    if suffixes or target_platforms or versioned:
+        return [
+            ":is_cp{}{}_{}".format(python_version, p, s)
+            for p in prefixes
+            for s in suffixes or [_non_versioned_platform(p) for p in target_platforms or []]
+        ], versioned
```

Comment:
Can you unroll this a bit? The nested list comps and `or`'d expression make this hard to read and I'm wary of remembering precedence correctly.

Reply:
OK, I will clean this up.